### DISCUSSION TITLE
CNF candidates for assessment

### DISCUSSION THREAD

**Discussion by vukg** [lfn-cnti/bestpractices#35] [General]
To get some practically relevant outcomes we will need to do assessment of number of real life CNFs. I am wondering who (CNF vendor / developer) would/could volunteer some to discuss here? Please specify the name, short description & developer.

Goal:
- Dialog with person that can benefit from using the best practice (eg. CNF Developer, Ops person)

**Comment by xmulligan** [comment]
Sorry I missed this for the meeting today! @taylor has been speaking with quite a few different developers and may have some suggestions. I'll also add it to the meeting for January 11th.

**Comment by taylor** [comment]
WIP list of [potential networking applications](https://github.com/cncf/cnf-conformance/blob/master/EXAMPLE-CNFS.md) from the CNF test suite.

**Comment by taylor** [comment]
Victor:  [Examples from CNF Testbed](https://github.com/cncf/cnf-testbed/tree/master/examples)

**Comment by taylor** [comment]
[PANTHEON.tech](https://pantheon.tech/) creates [CNFs](https://cdnf.io/). They have some examples on github:
- https://github.com/PANTHEONtech/cnf-examples
- https://github.com/PANTHEONtech/cnf-nsm

**Comment by xmulligan** [comment]
[Frederick] we could start with one, but we may need to expand to more in the future as different ones have different features and functionalities

**Reply by electrocucaracha** [reply]
@xmulligan I took one the GW-Tester CNF example used by the CNF-Testbed project and I improved implementing Multus, DANM and NSM CNIs. There is room to improve this one but maybe it can be used as starting point. https://github.com/electrocucaracha/gw-tester/

**Comment by vukg** [comment]
We could definitely start with these CNFs and use them as basis for validation of best practice aplicability and value add. 

However, what I had in mind are rather more complex solutions that could uncover a lot of edge cases. Let's say: https://mavenir.com/press-releases/mavenir-delivers-5g-core-user-plane-function-upf-with-nvidia-mellanox-smartnic/ or similar. How do we get vendors interested to contribute candidates?

**Reply by jambig** [reply]
Agree with @vukg 
We should be looking at CNFs like the Mavenir example, Nokia, Ericsson, Metaswitch, Cisco etc etc.

**Reply by rmerz** [reply]
I would as well support getting some CNF from an Ericsson or such. How should get them on board? 
@vukg I suspect we both work with Ericsson (for example). Should we approach them?

**Reply by vukg** [reply]
I already did, but the response is not fast.

on the other end if we are here specifying best practices that will end as mandatory in our (CSPs) RFQs, they will be more motivated. So I think we should focus on best practices for the time being and later on tests of concrete CNFs against those practices.
